### PR TITLE
Updated code from fo-dicom DicomElement addressing unit test failure related to Enum value handling

### PR DESCRIPTION
**Pull request by shivakumar-ms**
## Description
Updated DicomElement code from fo-dicom, handling the Enum value type.

## Related issues
[[AB#98797](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/98797)]

Reference PR in fo-dicom: https://github.com/fo-dicom/fo-dicom/pull/1441/

## Testing
Unit tests

### CODE REVIEW

File: forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs [hunk 1]
```diff
@@ -262,14 +262,27 @@ public override T Get<T>(int item = -1)
...
+    #region Private Members
+
+    private T GetValue<T>(TType val)
+    {
+        // If nullable, need to apply conversions on underlying type (#212)
+        var t = Nullable.GetUnderlyingType(typeof(T)) ?? typeof(T);
+
+        if (!t.IsEnum) return (T)Convert.ChangeType(val, t);
+
+        if (Enum.IsDefined(typeof(T), val)) return (T)Enum.ToObject(t, val);
```
Comment (wsugarman) [inline forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs:282]: `Enum.IsDefined` will throw an exception if the underlying `enum` type `T` and `val` are different types.

Comment (wsugarman) [reply forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs:282]: Your best bet may be to simply always call `ToString()` for `enum` types and call `Parse`. I don't like that very much... but this is a test util in a fork lol

File: forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs [hunk 2]
```diff
@@ -262,14 +262,27 @@ public override T Get<T>(int item = -1)
...
         }
 
         return base.Get<T>(item);
     }
 
     #endregion
+
+    #region Private Members
+
+    private T GetValue<T>(TType val)
```
[the remark this answers is not in the document]
Comment (shivakumar-ms) [reply forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs:275]: fo-dicom oss coding standard is to keep single return statements inline. So simply copied that change into our fork as is. I'll update this fork to comply with our standards.

[the remark this answers is not in the document]
Comment (shivakumar-ms) [reply forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs:275]: Done.

File: forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs (hunk 1 above)
Comment (shivakumar-ms) [reply forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs]: Done.

**Review by wsugarman — APPROVED** [review]

**Pull request merged**